plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -906,31 +934,275 @@ public function read_multiple( &$orders ) {
+		 *
+		 * In a case where post meta was updated (without updating post_modified date), post_modified would be equal to order_modified date.
+		 *
+		 * So we write back to the order table when order modified date is more recent than post modified date. Otherwise, we write to the post table.
+		 */
+		if ( $order_modified_date > $post_order_modified_date ) {
+			return;
+		} else {
+			$this->migrate_post_record( $order, $post_order );
+		}
+	}
+
+	/**
+	 * Get the post type order representation.
+	 *
+	 * @param \WP_Post $post Post object.
+	 *
+	 * @return \WC_Order Order object.
+	 */
+	private function get_cpt_order( $post ) {
+		$cpt_order = new \WC_Order();
+		$cpt_order->set_id( $post->ID );
+		$cpt_data_store = $this->get_cpt_data_store_instance();
+		$cpt_data_store->read( $cpt_order );
+		return $cpt_order;
+	}
+
+	/**
+	 * Helper function to get posts data for an order in bullk. We use to this to compute posts object in bulk so that we can compare it with COT data.
+	 *
+	 * @param array $order_ids List of order IDs.
+	 *
+	 * @return array List of posts.
+	 */
+	private function get_post_orders_for_ids( array $order_ids ): array {
+		$cpt_data_store = $this->get_cpt_data_store_instance();
+		// We have to bust meta cache, otherwise we will just get the meta cached by OrderTableDataStore.
+		foreach ( $order_ids as $order_id ) {
+			wp_cache_delete( WC_Order::generate_meta_cache_key( $order_id, 'orders' ), 'orders' );
+		}
+		$query_vars = array(
+			'include' => $order_ids,
+			'type'    => wc_get_order_types(),
+			'status'  => 'any',
+			'limit'   => count( $order_ids ),
+		);
+		$cpt_data_store->prime_caches_for_orders( $order_ids, $query_vars );
+		$orders = array();
+		foreach ( $order_ids as $order_id ) {
+			$order = new WC_Order();
 			$order->set_id( $order_id );
-			$order->read_meta_data();
-			$this->set_order_props_from_data( $order, $order_data );
-			$order->set_object_read( true );
+			$cpt_data_store->read( $order );
+			$orders[ $order_id ] = $order;
 		}
+		return $orders;
+	}
+
+	/**
+	 * Computes whether post has been updated after last order. Tries to do it as efficiently as possible.
+	 *
+	 * @param \WC_Abstract_Order $order Order object.
+	 * @param \WC_Abstract_Order $post_order Order object read from posts table.
+	 *
+	 * @return bool True if post is different than order.
+	 */
+	private function is_post_different_from_order( $order, $post_order ): bool {
+		if ( ArrayUtil::deep_compare_array_diff( $order->get_base_data(), $post_order->get_base_data(), false ) ) {
+			return true;
+		}
+
+		$meta_diff = $this->get_diff_meta_data_between_orders( $order, $post_order );
+		if ( ! empty( $meta_diff ) ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Migrate meta data from post to order.
+	 *
+	 * @param \WC_Order $order Order object.
+	 * @param \WC_Order $post_order Order object read from posts table.
```

Comment:
Maybe:

```suggestion
	 * @param \WC_Abstract_Order $order Order object.
	 * @param \WC_Abstract_Order $post_order Order object read from posts table.
```

Or the reverse (update the actual type hints)?

Reply:
This is correct, thanks!